xgcm/test/test_grid.py

```diff
@@ -14,600 +14,123 @@
-        ref = discontinuity
-
-    kw = {"boundary_discontinuity": discontinuity}
-    if right:
-        extended_raw = axis._extend_right(ds.XC, **kw)
-        extended = extended_raw[-1]
-        expected = ds.XC.data[0] + ref
-    else:
-        extended_raw = axis._extend_left(ds.XC, **kw)
-        extended = extended_raw[0]
-        expected = ds.XC.data[-1] - ref
-    assert isinstance(extended_raw, np.ndarray)
-    assert extended == expected
-
-
-@pytest.mark.parametrize("fill_value", [0, 10, 20])
-@pytest.mark.parametrize(
-    "boundary", ["fill", "extend", pytest.param("extrapolate", marks=pytest.mark.xfail)]
-)
-@pytest.mark.parametrize("periodic", [True, False])
-@pytest.mark.parametrize("is_left_edge", [True, False])
-@pytest.mark.parametrize("boundary_discontinuity", [None, 360])
-def test_get_edge_data(
-    periodic, fill_value, boundary, is_left_edge, boundary_discontinuity
-):
-    ds = datasets["1d_left"]
-    axis = Axis(ds, "X", periodic=periodic)
-    edge = axis._get_edge_data(
-        ds.XC,
-        boundary=boundary,
-        fill_value=fill_value,
-        is_left_edge=is_left_edge,
-        boundary_discontinuity=boundary_discontinuity,
-    )
-    if is_left_edge:
-        edge_periodic = ds.XC.data[-1]
-        if boundary_discontinuity is not None:
-            edge_periodic = edge_periodic - boundary_discontinuity
-        edge_extend = ds.XC.data[0]
-        edge_extra = ds.XC.data[0] - np.diff(ds.XC.data[0:2])
-    else:
-        edge_periodic = ds.XC.data[0]
-        if boundary_discontinuity is not None:
-            edge_periodic = edge_periodic + boundary_discontinuity
-        edge_extend = ds.XC.data[-1]
-        edge_extra = ds.XC.data[-1] + np.diff(ds.XC.data[-2:])
-    edge_fill = fill_value
-
-    if periodic:
-        assert edge_periodic == edge
-    else:
-        if boundary == "fill":
-            assert edge_fill == edge
-        elif boundary == "extend":
-            assert edge_extend == edge
-        elif boundary == "extrapolate":
-            assert edge_extra == edge
-        else:
-            assert 0
-
-
-def test_create_axis(all_datasets):
-    ds, periodic, expected = all_datasets
-    axis_objs = _get_axes(ds)
-    for ax_expected, coords_expected in expected["axes"].items():
-        assert ax_expected in axis_objs
-        this_axis = axis_objs[ax_expected]
-        for axis_name, coord_name in coords_expected.items():
-            assert axis_name in this_axis.coords
-            assert this_axis.coords[axis_name] == coord_name
-
-
 def _assert_axes_equal(ax1, ax2):
     assert ax1.name == ax2.name
     for pos, coord in ax1.coords.items():
         assert pos in ax2.coords
         assert coord == ax2.coords[pos]
     assert ax1._periodic == ax2._periodic
     assert ax1._default_shifts == ax2._default_shifts
-    assert ax1._facedim == ax2._facedim
     # TODO: make this work...
+    # assert ax1._facedim == ax2._facedim
     # assert ax1._connections == ax2._connections
```

Comment:
```suggestion
```

I guess this is what we had in mind with an `Axis.equals()` method. Either way this information is not stored in the axis anymore.

Reply:
Was the idea that we would have a `Grid.equals()`, that itself called `Axis.equals()`? Why were we going to have a grid.equals again? 

I think if we can't remember its fine to just leave equals out until it comes up again.

Reply:
I believe you remember that correctly. This would be useful, but is also pretty separable from this PR. I am going to open an issue to track this, and then we can remove the commented lines here and move on.

Reply:
#579